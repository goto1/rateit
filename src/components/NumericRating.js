import React from "react";
import PropTypes from "prop-types";
// import styled from "styled-components";

const NumericRating = ({ rating, fontSize }) => {
  let color = "#A1BA37";

  if (rating <= 3) {
    color = "#DEB21C";
  }
  if (rating <= 2) {
    color = "#D91948";
  }

  return (
    <div style={{ letterSpacing: "2px", fontSize: `${fontSize}px` }}>
      <span
        style={{
          color: color,
          fontSize: `${fontSize + 3}px`,
          fontWeight: "500"
        }}
      >
        {rating}
      </span>/5
    </div>
  );
};

NumericRating.propTypes = {
  rating: PropTypes.number.isRequired,
  fontSize: PropTypes.number.isRequired
};

// const NumericRating = ({ rating, fontSize }) => {
//   let color = "#A1BA37";

//   if (rating <= 3) {
//     color = "#DEB21C";
//   }
//   if (rating <= 2) {
//     color = "#D91948";
//   }

//   return (
//     <div style={{ letterSpacing: "2px" }}>
//       <span style={{ color, fontSize, fontWeight: "500" }}>{rating}</span>/5
//     </div>
//   );
// };

// NumericRating.propTypes = {
//   rating: PropTypes.number.isRequired,
//   fontSize: PropTypes.string.isRequired
// };

export default NumericRating;
