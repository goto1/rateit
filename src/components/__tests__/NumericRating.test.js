import React from "react";
import NumericRating from "../NumericRating";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import "jest-enzyme";
import "jest-styled-components";

describe("NumericRating component", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<NumericRating rating={4.5} />).toJSON();
    expect(tree).toMatchStyledComponentsSnapshot();
  });
});
