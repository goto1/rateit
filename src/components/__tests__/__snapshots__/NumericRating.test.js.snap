// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumericRating component should render correctly 1`] = `
.ecEnce {
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
}

.fxvkdv {
  font-weight: 500;
  color: #A1BA37;
}

<div
  className="sc-bdVaJa ecEnce"
>
  .fxvkdv {
  font-weight: 500;
  color: #A1BA37;
}

<span
    className="sc-bwzfXH fxvkdv"
    color="#A1BA37"
  >
    4.5
  </span>
  /5
</div>
`;
